From 48517d84c9976adeb71f0517ed02a79ad8ea445f Mon Sep 17 00:00:00 2001
From: Jerome Magnin <jmagnin@haproxy.com>
Date: Fri, 21 Feb 2020 10:33:12 +0100
Subject: [PATCH 01/25] MINOR: ist: add an iststop() function

Add a function that finds a character in an ist and returns an
updated ist with the length of the portion of the original string
that doesn't contain the char.

Might be backported to 2.1

(cherry picked from commit 9dde0b2d31dd0e588a6fcd34ca8b6dd6cee73e3d)
Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>
---
 include/common/ist.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/common/ist.h b/include/common/ist.h
index 10c0b6c08..0d8b45721 100644
--- a/include/common/ist.h
+++ b/include/common/ist.h
@@ -708,4 +708,16 @@ static inline struct ist istist(const struct ist ist, const struct ist pat)
 	return ist2(NULL, 0);
 }
 
+/*
+ * looks for the first occurence of <chr> in string <ist> and returns a shorter
+ * ist if char is found.
+ */
+static inline struct ist iststop(const struct ist ist, char chr)
+{
+	size_t len = 0;
+
+	while (len++ < ist.len && ist.ptr[len - 1] != chr)
+		;
+	return ist2(ist.ptr, len - 1);
+}
 #endif
-- 
2.22.0

